VocaDbWeb.Core/Helpers/ViewHelper.cs

```diff
@@ -207,7 +208,38 @@ public static RouteValueDictionary GetRouteValueDictionary(SongContract contract
 		/// Provides the content for individual table cells. Return value is raw HTML. Cannot be null.</param>
 		/// <returns></returns>
 		public static IHtmlContent Grid<T>(this IHtmlHelper htmlHelper, IEnumerable<T> items,
-			int columns, Func<T, IHtmlContent> contentFunc) => throw new NotImplementedException();
+			int columns, Func<T, IHtmlContent> contentFunc)
+		{
+			ParamIs.NotNull(() => htmlHelper);
+			ParamIs.NotNull(() => items);
+			ParamIs.Positive(columns, "columns");
+			ParamIs.NotNull(() => contentFunc);
+
+			var tableTag = new TagBuilder("table");
+			TagBuilder trTag = null;
+			int i = 0;
+
+			foreach (var item in items)
+			{
+				if (i % columns == 0)
+				{
+					if (trTag != null)
+						tableTag.InnerHtml.AppendHtml(trTag);
+
+					trTag = new TagBuilder("tr");
+				}
+
+				var tdTag = new TagBuilder("td");
+				tdTag.InnerHtml.AppendHtml(contentFunc(item));
+				trTag.InnerHtml.AppendHtml(tdTag);
+				i++;
+			}
+
+			if (trTag != null)
+				tableTag.InnerHtml.AppendHtml(trTag);
+
+			return tableTag;
+		}

```

Comment:
Replaced `InnerHtml +=` with `InnerHtml.AppendHtml`. Returns `TagBuilder` directly without converting to string.